include/swift/SILOptimizer/Utils/Generics.h

```diff
@@ -60,6 +60,46 @@ class ReabstractionInfo {
   /// SubstitutedType.
   CanSILFunctionType SpecializedType;
 
+  /// The generic environment to be used by the specialization.
+  GenericEnvironment *SpecializedGenericEnv;
+
+  /// The generic signature of the specializaiton.
+  /// It is nullptr if the specialization is not polymorphic.
+  GenericSignature *SpecializedGenSig;
+
+  // Set of the original substitutions used by the caller's
+  // apply instruction. It uses the contextual types of
+  // the caller.
+  ArrayRef<Substitution> ParamSubs;
+
+  // Specialized set of substitutions to be used by the caller
+  // when it calls a specialized function. It uses contextual
+  // types of the caller.
+  ArrayRef<Substitution> SpecializedParamSubs;
+
+  // The adjusted set of substitutions for the caller's apply
```

Comment:
It means that we modified the original set of substitutions from the apply instruction. The idea is that if the original substitution mapped to something generic, we now map the archetype to "itself".